### PR TITLE
Tx receipts & other headers

### PR DESCRIPTION
**Pull request by tudor-malene**
### Why is this change needed?
- for the UX we need transaction receipts, and in the future we'll need events
- the first part of including them is to add the root hashes to the header (this PR)
- the second part is saving them and exposing them via RPC ( future PR)

### What changes were made as part of this PR:

- functional PR 

- Add missing fields to the rollup header ( receipts hash, txHash, bloom)
- make the field names consistent with the ethereum block header ( the "number" field is now a *big.Int , and the state root is now called "Root")
- fix receipts logic ( there was a prepare stage that was missing)
- improve rollup verification logic ( when receiving a rollup verify that the fields in the header match the transactions)

### What are the key areas to look at
- nodecommon.types.go - the header
- enclave.go - where the fields get populated
- state.go - where the fields get verified

### CODE REVIEW

File: go/obscuronode/enclave/evm/evm_facade.go [hunk 1]
```diff
@@ -19,19 +19,20 @@ import (
 
 // ExecuteTransactions
 // header - the header of the rollup where this transaction will be included
-func ExecuteTransactions(txs []nodecommon.L2Tx, s *state.StateDB, header *nodecommon.Header, rollupResolver db.RollupResolver, chainID int64) map[common.Hash]*types.Receipt {
+// startAtCount - for the receipts and events, the evm needs to know for each transaction the order in which it was execute in the block.
```
Comment (BedrockSquirrel) [inline go/obscuronode/enclave/evm/evm_facade.go:22]: Found this param name/desc a bit confusing, we could maybe call it txIndex to match the naming in geth?

File: go/obscuronode/enclave/enclave.go [hunk 2]
```diff
@@ -171,7 +173,7 @@ func (e *enclaveImpl) start(block types.Block) {
 		// A new winner was found after gossiping. Start speculatively executing incoming transactions to already have a rollup ready when the next round starts.
 		case winnerRollup := <-e.roundWinnerCh:
 			hash := winnerRollup.Hash()
-			env.header = obscurocore.NewHeader(&hash, winnerRollup.Header.Number+1, e.config.HostID)
+			env.header = obscurocore.NewHeader(&hash, winnerRollup.Header.Number.Uint64()+1, e.config.HostID)
```
Comment (BedrockSquirrel) [inline go/obscuronode/enclave/enclave.go:176]: It's a bit annoying to have to get the `.Uint64()` all over the place. Would it make sense to make some of these height fields etc. a big int to match the header type?

**Review by BedrockSquirrel — COMMENTED** [review]
Couple of minor thoughts - generally LGTM

File: go/obscuronode/enclave/enclave.go (hunk 2 above)
Comment (tudor-malene) [reply go/obscuronode/enclave/enclave.go:176]: I know.. I tried somthing like that but the changes were quite extensive, so I gave up

File: go/obscuronode/enclave/evm/evm_facade.go (hunk 1 above)
Comment (tudor-malene) [reply go/obscuronode/enclave/evm/evm_facade.go:22]: fromTxIndex sounds better.

**Pull request merged**